src/eth/ethers/EthersEthereumService.ts

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright 2019 ShipChain, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { errors, ethers } from 'ethers';
+import { BigNumber, Network, shallowCopy, UnsignedTransaction } from 'ethers/utils';
+
+import { DeployedContractResult, EthereumService, TransactionEventHandlers } from '../EthereumService';
+import { JsonRpcProvider, Log } from 'ethers/providers';
+import { Logger } from '../../Logger';
+
+const config = require('config');
+
+const logger = Logger.get(module.filename);
+
+export class EthersEthereumService extends EthereumService {
+    private readonly provider: ethers.providers.Provider;
+
+    constructor() {
+        super();
+        if (config.has('GETH_NETWORK')) {
+            const network = config.get('GETH_NETWORK');
+            const applicableProviders: ethers.providers.BaseProvider[] = [];
+
+            logger.debug(`Connecting Ethers.js to [${network}]`);
+
+            // Add Infura provider if we have a projectId
+            if (config.has('INFURA_PROJECT_ID')) {
```

Comment:
I'd almost say lets axe this setting and make a regex that pulls the project ID out of GETH_NETWORK if it's an infura link? Thoughts?

Reply:
`GETH_NETWORK` settings are defined as the standard names of the network, `rinkeby`, `ropsten`,  `mainnet`.  I'd prefer to not extend those with extraneous data to parse for alternative purposes.  Especially since we can provide multiple both Etherscan and Infura tokens.